wa/commands/report.py

```diff
@@ -0,0 +1,288 @@
+                    try:
+                        select = int(input())
+                    except ValueError:
+                        print("Please select a valid path number")
+                        continue
+
+                    if select not in range(num_paths):
+                        print("Please select a valid path number")
+                        continue
+                    break
+
+                run_output = possible_outputs[select]
+
+            else:
+                run_output = possible_outputs[0]
+
+        rm = RunMonitor(run_output)
+        print(rm.generate_output(args.verbose))
+
+
+class RunMonitor:
+
+    @property
+    def elapsed_time(self):
+        if self._elapsed is None:
+            if self.ro.info.duration is None:
+                self._elapsed = datetime.utcnow() - self.ro.info.start_time
+            else:
+                self._elapsed = self.ro.info.duration
+        return self._elapsed
+
+    @property
+    def job_outputs(self):
+        if self._job_outputs is None:
+            self._job_outputs = {
+                (j_o.id, j_o.label, j_o.iteration): j_o for j_o in self.ro.jobs
+            }
+        return self._job_outputs
+
+    @property
+    def projected_duration(self):
+        elapsed = self.elapsed_time.total_seconds()
+        proj = timedelta(seconds=elapsed * (len(self.jobs) / len(self.segmented['finished'])))
+        return proj - self.elapsed_time
+
+    def __init__(self, ro):
+        self.ro = ro
+        self._elapsed = None
+        self._p_duration = None
+        self._job_outputs = None
+        self._termwidth = None
+        self._fmt = _simple_formatter()
+        self.get_data()
+
+    def get_data(self):
+        self.jobs = [state for label_id, state in self.ro.state.jobs.items()]
+        if self.jobs:
+            rc = self.ro.run_config
+            self.segmented = segment_jobs_by_state(self.jobs,
+                                                   rc.max_retries,
+                                                   rc.retry_on_status
+                                                   )
+
+    def generate_run_header(self):
+        info = self.ro.info
+
+        header = underline('Run Info')
+        header += "UUID: {}\n".format(info.uuid)
+        if info.run_name:
+            header += "Run name: {}\n".format(info.run_name)
+        if info.project:
+            header += "Project: {}\n".format(info.project)
+        if info.project_stage:
+            header += "Project stage: {}\n".format(info.project_stage)
+
+        if info.start_time:
+            duration = _seconds_as_smh(self.elapsed_time.total_seconds())
+            header += ("Start time: {}\n"
+                       "Duration: {:02}:{:02}:{:02}\n"
+                       ).format(info.start_time,
+                                duration[2], duration[1], duration[0],
+                                )
+            if len(self.segmented['finished']) > 0 and not info.end_time:
```

Comment:
C:157,15: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
https://travis-ci.org/github/ARM-software/workload-automation/jobs/706972953